internal/command/cliconfig/provider_installation.go

```diff
@@ -246,7 +246,7 @@ func decodeProviderInstallationFromConfig(hclFile *hclast.File) ([]*ProviderInst
 						diags = diags.Append(tfdiags.Sourceless(
 							tfdiags.Error,
 							"Invalid provider installation dev overrides",
-							fmt.Sprintf("The entry %q in %s is not a valid provider source string.", rawAddr, block.Pos()),
+							fmt.Sprintf("The entry %q in %s is not a valid provider source string.\n%s", rawAddr, block.Pos(), moreDiags.Err().Error()),
```

Comment:
Based on the example in the PR description, I think this might look better with two newlines before the interpolated error. For example:

```
╷
│ Error: Invalid provider installation dev overrides
│
│ The entry "registry.terraform.io/joneshf/terraform-provider-foo" in 1:1 is
│ not a valid provider source string.
│
│ Invalid provider type: Provider source "joneshf/terraform-provider-foo" has
│ a type with the prefix "terraform-provider-", which isn't valid. Although
│ that prefix is often used in the names of version control repositories for
│ Terraform providers, provider source strings should not include it.
│
│ Did you mean "joneshf/foo"?
╵
```